tests/main.test.ts

```diff
@@ -2445,4 +2471,81 @@ describe('zip-it-and-ship-it', () => {
     expect(result.stderr).not.toContain('Dynamic require of "path" is not supported')
     expect(result).not.toBeInstanceOf(Error)
   })
+
+  testMany(
+    'bundlers emit entry file with cjs extension when FF on',
+    ['bundler_default', 'bundler_esbuild', 'bundler_nft'],
+    async (options) => {
+      const fixtureName = 'node-esm'
+      const opts = merge(options, {
+        basePath: join(FIXTURES_DIR, fixtureName),
+        featureFlags: { zisi_output_cjs_extension: true },
+      })
+      const { files, tmpDir } = await zipFixture(fixtureName, {
+        length: 2,
+        opts,
+      })
+
+      await unzipFiles(files)
+
+      const bundledFile2 = await importFunctionFile(join(tmpDir, 'func2.cjs'))
+      expect(bundledFile2.handler()).toBe(true)
+
+      if (options.getCurrentBundlerName() === 'esbuild') {
+        // nft does not create an entry here because the main file is already an entrypoint
+        const bundledFile1 = await importFunctionFile(join(tmpDir, 'func1.cjs'))
+        expect(bundledFile1.handler()).toBe(true)
+      }
+    },
+  )
+
+  testMany('bundlers keep cjs file extension', ['bundler_default', 'bundler_nft'], async (options) => {
+    const fixtureName = 'node-cjs-extension'
+    const opts = merge(options, {
+      basePath: join(FIXTURES_DIR, fixtureName),
+    })
+    const { files, tmpDir } = await zipFixture(fixtureName, {
+      length: 3,
+      opts,
+    })
+
+    await unzipFiles(files)
+
+    const bundledFile1 = await importFunctionFile(join(tmpDir, 'func1.cjs'))
+    const bundledFile2 = await importFunctionFile(join(tmpDir, 'func2.cjs'))
+    const bundledFile3 = await importFunctionFile(join(tmpDir, 'index.cjs'))
+
+    expect(bundledFile1.handler()).toBe(true)
+    expect(bundledFile2.handler()).toBe(true)
+    expect(bundledFile3.handler()).toBe(true)
+  })
+
+  testMany(
+    'bundlers do not create cjs entry file if entry with js extension is already present',
```

Comment:
Nit for consistency.

```suggestion
    'Does not create .cjs entry file if entry with .js extension is already present',
```